packages/web-frontend/src/components/DateRangePicker/DateRangePicker.js

```diff
@@ -116,25 +118,36 @@ export const DateRangePicker = ({
   const isSingleDate = GRANULARITIES_WITH_ONE_DATE.includes(granularity);
   const { momentShorthand } = GRANULARITY_CONFIG[granularity];
 
-  const minMomentDate = moment(MIN_DATE_PICKER_DATE);
-  const maxMomentDate = moment();
-
-  const defaultStartDate = isSingleDate ? moment() : minMomentDate;
-  const defaultEndDate = isSingleDate ? defaultStartDate : maxMomentDate;
+  const minMomentDate = min ? moment(min) : moment(MIN_DATE_PICKER_DATE);
+  const maxMomentDate = max ? moment(max) : moment();
+
+  let defaultStartDate;
+  let defaultEndDate;
+  if (isSingleDate) {
+    defaultStartDate = moment();
+    if (defaultStartDate < minMomentDate) defaultStartDate = minMomentDate;
+    if (defaultStartDate > maxMomentDate) defaultStartDate = maxMomentDate;
```

Comment:
Can you pull out a util called `constrainDate(defaultStartDate, minMomentDate, maxMomentDate)`? I reckon I'd find it more readable to have that as a named concept than as a list of conditions inline (especially as it's used in the next block too)

Reply:
Extracted